evalml/pipelines/components/transformers/imputers/simple_imputer.py

```diff
@@ -78,11 +78,25 @@ def fit(self, X, y=None):
             raise ValueError(
                 "SimpleImputer cannot handle dataframes with both boolean and categorical features.  Use Imputer instead.",
             )
+
         nan_ratio = X.isna().sum() / X.shape[0]
         self._all_null_cols = nan_ratio[nan_ratio == 1].index.tolist()
 
         X, _ = drop_natural_language_columns(X)
-        X = set_boolean_columns_to_categorical(X)
+
+        # Convert any boolean columns to IntegerNullable, but keep track of the columns so they can be converted back
+        self._boolean_cols = X.ww.schema._filter_cols(
+            include=["Boolean", "BooleanNullable"],
+        )
```

Comment:
I might be being pedantic, but is the preferred way to call a private function on the schema?  I thought there was a select function on the ww accessor?

Reply:
Certainly. I stole this line directly from `set_boolean_columns_to_integer`, but I can switch both places over to use `select` instead.